DataStructureProject/LinkedList.rb

```diff
@@ -0,0 +1,101 @@
+require_relative 'DataStructure'
+
+class ListNode < Node
+
+        attr_accessor :next
+
+        def initialize(value = nil, nxt = nil)
+                super value
+                @next = nxt
+        end
+end
+
+class LinkedList < DataStructures
+
+        def initialize
+                @head = nil
+                @tail = nil
+        end
+
+        def insert(value)
+                new_node = ListNode.new(value)
+                
+                if @head.nil?
+                        @head = new_node
+                        @tail = new_node
+                else
+                        @tail.next = new_node
+                        @tail = new_node
+                end
+        end
+        
+        def remove(value)
+                return if @head.nil?
+
+                if @head.value == value
+                        @head = @head.next
+                        return
+                end
+
+                prev = nil
+                curr_node = @head
+                while !curr_node.nil? and curr_node.value != value
+                        prev = curr_node
+                        curr_node = curr_node.next
+                end
+                return if curr_node.nil?
+                prev.next = curr_node.next
+                @tail = prev if curr_node == @tail
+        end
```

Comment:
Try something like this
```suggestion
        def remove(value)
                return if @head.nil?
                if @head.value == value
                        @head = @head.next
                else
                        prev = nil
                        curr_node = @head
                        while !curr_node.nil?
                                if curr_node.value == value
                                       prev.next = curr_node.next
                                       @tail = prev if curr_node == @tail
                                       break
                                end
                                prev = curr_node
                                curr_node = curr_node.next
                        end
                end
        end
```